scio-parquet/src/main/scala/com/spotify/scio/parquet/GcsConnectorUtil.scala

```diff
@@ -17,33 +17,52 @@
 
 package com.spotify.scio.parquet
 
+import com.google.auth.oauth2.{GoogleCredentials, ServiceAccountCredentials}
+import com.google.cloud.hadoop.util.AccessTokenProvider
 import com.spotify.scio.ScioContext
 import com.spotify.scio.util.ScioUtil
+import org.apache.hadoop.conf.Configuration
 import org.apache.hadoop.mapreduce.Job
 import org.apache.hadoop.mapreduce.lib.input.FileInputFormat
 
+import java.io.File
+import java.util.Locale
+import scala.util.Try
+
 private[parquet] object GcsConnectorUtil {
-  def setCredentials(job: Job): Unit =
+  def setCredentials(job: Job): Unit = {
     // These are needed since `FileInputFormat.setInputPaths` validates paths locally and
     // requires the user's GCP credentials.
-    sys.env.get("GOOGLE_APPLICATION_CREDENTIALS") match {
-      case Some(json) =>
-        job.getConfiguration
-          .set("fs.gs.auth.service.account.json.keyfile", json)
-      case None =>
-        // Client id/secret of Google-managed project associated with the Cloud SDK
+    lazy val applicationDefault = Try(GoogleCredentials.getApplicationDefault())
+
+    sys.env
+      .get("GOOGLE_APPLICATION_CREDENTIALS")
+      .filter(_.nonEmpty)
+      .orElse {
+        applicationDefault.toOption match {
+          case Some(_: ServiceAccountCredentials) => getWellKnownCredentialFile.map(_.toString)
+          case _                                  => None
+        }
+      } match {
+      case Some(serviceAccountKeyFile) =>
+        job.getConfiguration.set("fs.gs.auth.service.account.json.keyfile", serviceAccountKeyFile)
+      case _ if applicationDefault.isSuccess => // ADC exists but isn't a service account
+        job.getConfiguration.set(
+          "fs.gs.auth.access.token.provider.impl",
+          "com.spotify.scio.parquet.ApplicationDefaultTokenProvider"
+        )
+      case _ =>
         job.getConfiguration
           .setBoolean("fs.gs.auth.service.account.enable", false)
-        job.getConfiguration.set("fs.gs.auth.client.id", "32555940559.apps.googleusercontent.com")
-        job.getConfiguration
-          .set("fs.gs.auth.client.secret", "ZmssLNjJy2998hD4CTg2ejr2")
+        job.getConfiguration.unset("fs.gs.auth.null.enable")
```

Comment:
this option is required to be set to produce a valid config in this case: https://github.com/GoogleCloudDataproc/hadoop-connectors/blob/28c40997412991a16bff0e512c6dc6ed9c606874/util/src/main/java/com/google/cloud/hadoop/util/CredentialOptions.java#L111

it's only meant to be used for unit testing, as there are no credentials. (it will log a warning about this on job launch as well)

Reply:
ah this should be `.setBoolean` to true, somehow got messed up refactoring 😅